r/R/compute.R

```diff
@@ -284,3 +284,14 @@ cast_options <- function(safe = TRUE, ...) {
   )
   modifyList(opts, list(...))
 }
+
+#' @export
+strsplit <- function(x, split, fixed = TRUE, perl = FALSE, useBytes = FALSE, ...) UseMethod("strsplit")
+
+#' @export
+strsplit.default <- base::strsplit
```

Comment:
Since `strsplit()` is not a generic, this is the only way we can make it usable in standard evaluation. But this will generate an additional message when Arrow loads:
```
The following object is masked from ‘package:base’:

    strsplit
```
I believe we want to avoid masking any more functions like this—is that correct @nealrichardson?

The alternative is to make `strsplit()` usable in dplyr. Non-standard evaluation allows us to do that without any masking issues. To do that, you add a mapping here: https://github.com/apache/arrow/blob/master/r/R/dplyr.R#L396

Reply:
Correct, we don't want to mask. We should just wire this up in the dplyr verbs as @ianmcook suggests. `nchar()` on L448 there could be a good model to follow, or perhaps sub/gsub if you need more complexity. Likewise, we'll want to add something for the stringr versions of this (str_split, str_split_fixed?) too.

Reply:
Yep, let's sync up when you're ready @thisisnic and I can give some tips on how to do the stringr versions, like we discussed on Slack 👍

Reply:
P.S. some of the automated tests are failing because of warnings that there is no docs entry for `strsplit`. These warnings will go away after you wire this up to work only inside dplyr verbs

Reply:
Thanks for the feedback, will give you a shout Ian!